test/test.h

```diff
@@ -53,6 +54,23 @@ make_config(const std::string& path_prefix,
             const std::unordered_map<std::string, std::string>& overrides = {},
             const std::unordered_set<std::string>& removes = {});
 
+template <typename container_t>
+testing::AssertionResult shape_equality(const container_t& expected,
+                                        const container_t& actual,
+                                        typename container_t::value_type::first_type tolerance = 1) {
+  auto hd = Polyline2<typename container_t::value_type>::HausdorffDistance(expected, actual);
+  if (hd > tolerance)
+    return testing::AssertionFailure() << "shape exceeds tolerance by " << hd - tolerance;
+  return testing::AssertionSuccess();
+}
+
+inline testing::AssertionResult
+encoded_shape_equality(const std::string& expected, const std::string& actual, double tolerance = 1) {
```

Comment:
added these two helper functions for any time we need to compare shape in a robust way